server/src/internalClusterTest/java/org/elasticsearch/timeseries/support/TimeSeriesDimensionsLimitIT.java

```diff
@@ -0,0 +1,200 @@
+        source.put("gauge", randomIntBetween(10, 20));
+        source.put("@timestamp", Instant.now().toEpochMilli());
+        for (int i = 0; i < dimensionFieldLimit; i++) {
+            source.put(dimensionFieldNames.get(i), randomAlphaOfLength(1024));
+        }
+        final IndexResponse indexResponse = client().prepareIndex("test").setSource(source).get();
+        assertEquals(RestStatus.CREATED.getStatus(), indexResponse.status().getStatus());
+    }
+
+    public void testTotalDimensionFieldsSizeLuceneLimitPlusOne() throws IOException {
+        int dimensionFieldLimit = 22;
+        final List<String> dimensionFieldNames = new ArrayList<>();
+        createTimeSeriesIndex(mapping -> {
+            for (int i = 0; i < dimensionFieldLimit; i++) {
+                String dimensionFieldName = randomAlphaOfLength(512);
+                dimensionFieldNames.add(dimensionFieldName);
+                mapping.startObject(dimensionFieldName).field("type", "keyword").field("time_series_dimension", true).endObject();
+            }
+        },
+            mapping -> mapping.startObject("gauge").field("type", "integer").field("time_series_metric", "gauge").endObject(),
+            () -> List.of(dimensionFieldNames.get(0)),
+            dimensionFieldLimit
+        );
+
+        final Map<String, Object> source = new HashMap<>();
+        source.put("routing_field", randomAlphaOfLength(1024));
+        source.put("gauge", randomIntBetween(10, 20));
+        source.put("@timestamp", Instant.now().toEpochMilli());
+        for (int i = 0; i < dimensionFieldLimit; i++) {
+            source.put(dimensionFieldNames.get(i), randomAlphaOfLength(1024));
+        }
+        final Exception ex = expectThrows(DocumentParsingException.class, () -> client().prepareIndex("test").setSource(source).get());
+        assertEquals("_tsid longer than [32766] bytes [33903].", ex.getCause().getMessage());
+    }
+
+    private void createTimeSeriesIndex(
+        final CheckedConsumer<XContentBuilder, IOException> dimensions,
+        final CheckedConsumer<XContentBuilder, IOException> metrics,
+        final Supplier<List<String>> routingPaths,
+        int dimensionsFieldLimit
+    ) throws IOException {
+        XContentBuilder mapping = JsonXContent.contentBuilder();
+        mapping.startObject().startObject("properties");
+        mapping.startObject("@timestamp").field("type", "date").endObject();
+        metrics.accept(mapping);
+        dimensions.accept(mapping);
+        mapping.endObject().endObject();
+
+        Settings.Builder settings = Settings.builder()
+            .put(IndexSettings.MODE.getKey(), IndexMode.TIME_SERIES)
+            .putList(IndexMetadata.INDEX_ROUTING_PATH.getKey(), routingPaths.get())
+            .put(IndexSettings.TIME_SERIES_START_TIME.getKey(), "2000-01-08T23:40:53.384Z")
+            .put(IndexSettings.TIME_SERIES_END_TIME.getKey(), "2106-01-08T23:40:53.384Z");
+
+        if (dimensionsFieldLimit > 0) {
```

Comment:
maybe make `dimensionsFieldLimit` `java.lang.Integer` and have a test which passes dow null?